src/Aspire.Hosting.Azure/AzureResourceExtensions.cs

```diff
@@ -196,17 +196,25 @@ public static IResourceBuilder<AzureStorageEmulatorResource> UsePersistence(this
     /// For more information on the Azure Cosmos DB emulator, see <a href="https://learn.microsoft.com/azure/cosmos-db/emulator#authentication"></a>
     /// </summary>
     /// <param name="builder">The Azure Cosmos DB resource builder.</param>
-    /// <param name="port">The port used for the client SDK to access the emulator. Defaults to <c>8081</c></param>
-    /// <param name="imageTag">The image tag for the <c>mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator</c> image.</param>
+    /// <param name="configureContainer">Callback that exposes underlying container used for emulation to allow for customization.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
     /// <remarks>
     /// When using the Azure Cosmos DB emulator, the container requires a TLS/SSL certificate.
     /// For more information, see <a href="https://learn.microsoft.com/azure/cosmos-db/how-to-develop-emulator?tabs=docker-linux#export-the-emulators-tlsssl-certificate"></a>
     /// </remarks>
-    public static IResourceBuilder<AzureCosmosDBResource> UseEmulator(this IResourceBuilder<AzureCosmosDBResource> builder, int? port = null, string? imageTag = null)
+    public static IResourceBuilder<AzureCosmosDBResource> UseEmulator(this IResourceBuilder<AzureCosmosDBResource> builder, Action<IResourceBuilder<AzureCosmosDBEmulatorResource>>? configureContainer = null)
```

Comment:
File an issue for fixing the host port. This is important for using external tools with containers (azure data studio for e.g.).

Reply:
I think that this issue will address that:
https://github.com/dotnet/aspire/issues/2169

I'll add a note to add extension methods for the Azure Storage and Azure Cosmos container resource surrogates to make it more discoverable in those cases.